lib/AST/Identifier.cpp

```diff
@@ -21,12 +21,9 @@
 #include "clang/Basic/CharInfo.h"
 using namespace swift;
 
-void *DeclBaseName::SubscriptIdentifierData =
-    &DeclBaseName::SubscriptIdentifierData;
-void *DeclBaseName::ConstructorIdentifierData =
-    &DeclBaseName::ConstructorIdentifierData;
-void *DeclBaseName::DestructorIdentifierData =
-    &DeclBaseName::DestructorIdentifierData;
+Identifier::Aligner DeclBaseName::SubscriptIdentifierData{};
+Identifier::Aligner DeclBaseName::ConstructorIdentifierData{};
+Identifier::Aligner DeclBaseName::DestructorIdentifierData{};
```

Comment:
Bonus nitpick: these should probably be both `const` and `constexpr` to make sure they don't cause any static constructors.